mycroft/skills/mycroft_skill/mycroft_skill.py

```diff
@@ -157,6 +162,29 @@ def __init__(self, name=None, bus=None, use_settings=True):
         self.event_scheduler = EventSchedulerInterface(self.name)
         self.intent_service = IntentServiceInterface()
 
+    def _init_settings(self, prefix):
+        """Setup skill settings."""
+
+        #: First, check the skill directory to not break existing setups
+        self.settings_dir = self.root_dir
+
+        #: Then, check XDG_CONFIG_DIR
+        if not isdir(self.settings_dir):
+            for dir in BaseDirectory.load_config_paths('mycroft'):
+                path = dir + 'skills' + basename(self.root_dir)
+                #: If there is a settings file here, use it
+                if isdir(path):
```

Comment:
On first start will this path actually exist? shouldn't it be created if the XDG_CONFIG_DIR is to be used?

Reply:
It might not exist, but that shouldn't be a problem as afaik the skills currently start without settings as well on first start. At least on my system this works fine on first start without any settings anywhere.

When writing the settings we do make sure XDG_CONFIG_DIR exists though.